modules/openapi-generator/src/main/resources/typescript-fetch/api.mustache

```diff
@@ -1,364 +1,442 @@
+     * fail when the string isn't actually JSON.
+     */
+    protected getResponseType(returnType: string): ResponseType {
+        switch (returnType) {
+            case 'string':
+                return 'text'
+            default:
+                return 'JSON'
         }
     }
+
+    /**
+     * Create a shallow clone of `this` by constructing a new instance
+     * and then shallow cloning data members.
+     */
+    private clone<T extends BaseAPI>(this: T): T {
+        const constructor = this.constructor as any;
+        const next = new constructor(this.configuration);
+        next.middleware = this.middleware.slice();
+        return next;
+    }
 };
 
-/**
- * 
- * @export
- * @class RequiredError
- * @extends {Error}
- */
 export class RequiredError extends Error {
-    name: "RequiredError"
+    name = 'RequiredError';
     constructor(public field: string, msg?: string) {
         super(msg);
     }
 }
 
-{{#models}}
-{{#model}}{{#isEnum}}{{>modelEnum}}{{/isEnum}}{{^isEnum}}{{>modelGeneric}}{{/isEnum}}{{/model}}
-{{/models}}
 {{#apiInfo}}{{#apis}}{{#operations}}
 /**
- * {{classname}} - fetch parameter creator{{#description}}
- * {{&description}}{{/description}}
- * @export
+ * {{#description}}{{{description}}}{{/description}}{{^description}}no description{{/description}}
  */
-export const {{classname}}FetchParamCreator = function (configuration?: Configuration) {
-    return {
+export class {{classname}} extends BaseAPI {
+
     {{#operation}}
-        /**
-         * {{&notes}}
-         {{#summary}}
-         * @summary {{&summary}}
-         {{/summary}}
-         {{#allParams}}
-         * @param {{=<% %>=}}{<%&dataType%>}<%={{ }}=%> {{^required}}[{{/required}}{{paramName}}{{^required}}]{{/required}} {{description}}
-         {{/allParams}}
-         * @param {*} [options] Override http request option.
-         * @throws {RequiredError}
-         */
-        {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}options: any = {}): FetchArgs {
-    {{#allParams}}
-    {{#required}}
-            // verify required parameter '{{paramName}}' is not null or undefined
-            if ({{paramName}} === null || {{paramName}} === undefined) {
-                throw new RequiredError('{{paramName}}','Required parameter {{paramName}} was null or undefined when calling {{nickname}}.');
-            }
-    {{/required}}
-    {{/allParams}}
-            const localVarPath = `{{{path}}}`{{#pathParams}}
-                .replace(`{${"{{baseName}}"}}`, encodeURIComponent(String({{paramName}}))){{/pathParams}};
-            const localVarUrlObj = url.parse(localVarPath, true);
-            const localVarRequestOptions = Object.assign({ method: '{{httpMethod}}' }, options);
-            const localVarHeaderParameter = {} as any;
-            const localVarQueryParameter = {} as any;
-    {{#hasFormParams}}
-            const localVarFormParams = new url.URLSearchParams();
```

Comment:
@TiFu my fork does away with the dependency on the URL module and uses browser native APIs such as `FormData` and `encodeURIComponent`.

This solves issues such as https://github.com/swagger-api/swagger-codegen/issues/6403

From memory some users encountered issues with `@types` definitions for isomorphic fetch as well, but I can't find the bug reports :/